dev/breeze/SELECTIVE_CHECKS.md

```diff
@@ -103,51 +103,57 @@ The logic implements the following rules:
 * if `Image building` is disabled, only basic pre-commits are enabled - no 'image-depending` pre-commits
   are enabled.
 * If there are some setup files changed, `upgrade to newer dependencies` is enabled.
+* If docs are build, the `docs-filter` will determine which docs packages to build. This is based on
+  several criteria: if any of the airflow core, charts, docker-stack, providers files or docs have changed,
```

Comment:
We may have cross refence between peoviders.
For example amazon provider uses common.sql example dags as part of doc build.

Reply:
Yes. That is included. See the tests on google provider.

Reply:
We are using automatically generated "generated/provider-dependencies.json" to make sure that for example when any of the google docs are changed, any provider that depends on it is also added to built docs packages

Reply:
```
       pytest.param(
            ("docs/apache-airflow-providers-google/docs.rst",),
            {
                "docs-filter": "--package-filter apache-airflow-providers-amazon "
                "--package-filter apache-airflow-providers-apache-beam "
                "--package-filter apache-airflow-providers-apache-cassandra "
                "--package-filter apache-airflow-providers-cncf-kubernetes "
                "--package-filter apache-airflow-providers-common-sql "
                "--package-filter apache-airflow-providers-facebook "
                "--package-filter apache-airflow-providers-google "
                "--package-filter apache-airflow-providers-hashicorp "
                "--package-filter apache-airflow-providers-microsoft-azure "
                "--package-filter apache-airflow-providers-microsoft-mssql "
                "--package-filter apache-airflow-providers-mysql "
                "--package-filter apache-airflow-providers-oracle "
                "--package-filter apache-airflow-providers-postgres "
                "--package-filter apache-airflow-providers-presto "
                "--package-filter apache-airflow-providers-salesforce "
                "--package-filter apache-airflow-providers-sftp "
                "--package-filter apache-airflow-providers-ssh "
                "--package-filter apache-airflow-providers-trino",
            },
            id="Google provider docs changed",
        ),
```

Reply:
I've added a fixup to SELECTIVE_CHECKS.rst to make sure it is crystal-clear in the docs.

Reply:
I also added another test case - speecifically for common-sql, to show up what happens when any of the common/sql python files change (this case is also handled - because changes in any of the python files in the dependent provider might impact generated documentation due to auto-api generated documentation from docstrings.

But the change in selective_checks I implemented is nicely handking this case too


```
        pytest.param(
            ("airflow/providers/common/sql/common_sql_python.py",),
            {
                "docs-filter": "--package-filter apache-airflow "
                "--package-filter apache-airflow-providers-amazon "
                "--package-filter apache-airflow-providers-apache-drill "
                "--package-filter apache-airflow-providers-apache-druid "
                "--package-filter apache-airflow-providers-apache-hive "
                "--package-filter apache-airflow-providers-apache-impala "
                "--package-filter apache-airflow-providers-apache-pinot "
                "--package-filter apache-airflow-providers-common-sql "
                "--package-filter apache-airflow-providers-databricks "
                "--package-filter apache-airflow-providers-elasticsearch "
                "--package-filter apache-airflow-providers-exasol "
                "--package-filter apache-airflow-providers-google "
                "--package-filter apache-airflow-providers-jdbc "
                "--package-filter apache-airflow-providers-microsoft-mssql "
                "--package-filter apache-airflow-providers-mysql "
                "--package-filter apache-airflow-providers-odbc "
                "--package-filter apache-airflow-providers-oracle "
                "--package-filter apache-airflow-providers-postgres "
                "--package-filter apache-airflow-providers-presto "
                "--package-filter apache-airflow-providers-qubole "
                "--package-filter apache-airflow-providers-slack "
                "--package-filter apache-airflow-providers-snowflake "
                "--package-filter apache-airflow-providers-sqlite "
                "--package-filter apache-airflow-providers-trino "
                "--package-filter apache-airflow-providers-vertica",
            },
            id="Common SQL provider package python files changed changed",
        ),
```

Reply:
BTW. The whole PR here is based on the observation, that there is no way provider documentation depends on a different provider, if the provider itsefl does not depend on it (i.e. imports the other provider) - we already have that information and we are making good use of it. 

Of course you could like make a reference from provider A to B is A does not depend on B, but that would be rather stupid to do and we would have found it in the "canary" build which always runs "full" build regardless.

There is an updated tests that makes sure this is happening too (`docs-filter: ""`  means that all docs packages are built):

```
        pytest.param(
            ("INTHEWILD.md",),
            (),
            "main",
            {
                "all-python-versions": "['3.7', '3.8', '3.9', '3.10']",
                "all-python-versions-list-as-string": "3.7 3.8 3.9 3.10",
                "image-build": "true",
                "needs-helm-tests": "true",
                "run-tests": "true",
                "docs-build": "true",
                "docs-filter": "",
                "upgrade-to-newer-dependencies": "true",
                "parallel-test-types": "Core Providers[-amazon,google] Other Providers[amazon] WWW "
                "API Always CLI Providers[google]",
            },
            id="All tests run on push even if unimportant file changed",
```